app/src/main/java/com/lagradost/cloudstream3/utils/ImageModuleCoil.kt

```diff
@@ -40,23 +41,23 @@ object ImageLoader {
             .build()
 
         return ImageLoader.Builder(context)
-            .crossfade(true)
+            .crossfade(250)
             .respectCacheHeaders(false)
             /** !Only use default placeholders and errors, if not using this instance for local
              * image buttons because when animating this will appear or in more cases **/
             //.placeholder(R.drawable.logo)
             //.error(R.drawable.logo)
-            .allowHardware(true)
+            .allowHardware(false) // takes a toll on battery (only allow if app is like instagram or photos)
```

Comment:
Is this really the case? I thought hardware bitmaps enabled the hardware to save energy by "offloading the work to dedicated co-processors". Can you link where you got this information or explain more in detail.

Reply:
the theory is hw decoding works well only in devices that are recent maybe like sdk int > 28, and it is only needed to render large bitmaps.
actually we are fucked instacart released coil 3 in which we dont need our own singleton instead we should inherit their factory in application class